src/backup.c

```diff
@@ -2255,12 +2255,12 @@ init_scan_bins(char *bin_list, as_scan *scan)
 	as_vector bin_vec;
 	as_vector_inita(&bin_vec, sizeof (void *), 25);
 
-	if (bin_list[0] == 0) {
+	if (clone[0] == 0) {
 		err("Empty bin list");
 		goto cleanup1;
 	}
 
-	split_string(bin_list, ',', true, &bin_vec);
+	split_string(clone, ',', true, &bin_vec);
```

Comment:
Is the problem that split_string mangles the bin_list or that the threads mangle the bin_list because pointers to bin_list are stored in the bin_vec?

Reply:
Kind of both, each thread has a pointer to the same list, and split_string mangles them. So each thread ends up with a different view of the string when they parse it.